src/resources/cpp/lambdas.md

```diff
@@ -0,0 +1,244 @@
+# Lambdas
+
+## What is a lambda?
+
+Lambdas are similar to functions, however they are not identical they specifically are
+[closures](<https://en.wikipedia.org/wiki/Closure_(computer_programming)>).
+
+if they are similar to functions one might ask why would I ever need a lambda? For example, algorithms in the standard
+library can often have their behavior changed using something callable. Creating a whole function is often excessive,
+and we use lambdas instead:
+
+`int non_a_amount = std::ranges::count_if("abcaabbac", [](char c) { return c != 'a'; });`
+
+This counts all results where the lambda returns true, in this case if the latter is not 'a'.
+
+Another example is where we want to print each value in an array:
+
+```cpp
+#include <algorithm>
+#include <iostream>
+#include <vector>
+
+int main()
+{
+    std::vector<int> v{3, -4, 2, -8, 15, 267};
+
+    // Here we declare a local lambda object, called print
+    auto print = [](const int& n) { std::cout << n << '\n'; };
+
+    // Here we use this lambda object in the algorithm for each, where the lambda will be called for each
+    // iteration and print the value in the vector.
+    std::for_each(v.cbegin(), v.cend(), print);
+
+    // As a lambda is similar to a function we can also call it like a function. This will print 5
+    print(5);
+
+    return 0;
+}
+```
+
+## Structure of lambdas
+
+A lambda consists of at least three parts:
+
+- `[]`, the introducer, containing captures.
+- `()`, the parameters (optional in some cases).
+- `{}`, the body of the lambda.
+
+There are many further optional parts, but `[](){}` or `[]{}` are the bare minimum, depending on the language version.
+
+for a lambda the return type does not need to be specified like functions do. The return type is deduced by the
```

Comment:
Fixing the grammar:

For a lambda, the return type does not need to be explicitly specified, unlike functions. The return type is deduced by the compiler. However, you may specify a trailing return type to the lambda explicitly.